internal/core/data/device_test.go

```diff
@@ -108,7 +108,7 @@ func newMockDeviceClient() *mocks.DeviceClient {
 	})).Return(mockDeviceResultFn, nil)
 	client.On("Device", mock.MatchedBy(func(id string) bool {
 		return id == "404"
-	})).Return(mockDeviceResultFn, types.ErrNotFound{})
+	})).Return(mockDeviceResultFn, types.NewErrServiceClient(404, []byte{}))
```

Comment:
Please, replace all 404 by `http.StatusNotFound`

Reply:
Thank you